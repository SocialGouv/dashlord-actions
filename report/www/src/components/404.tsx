import * as React from "react";

import Table from "@codegouvfr/react-dsfr/Table";

import { Panel } from "./Panel";

type Report404Props = { data: Wget404Report };

export const Report404: React.FC<Report404Props> = ({ data }) => {
  const tableData = data && [
    ["destination"],
    ...data.map((url2) => [url2.link]),
  ];
  return (
    (data && data.length && (
      <Panel title="Erreurs 404" info="Pages introuvables">
        <Table data={tableData} />
      </Panel>
    )) ||
    null
  );
};
