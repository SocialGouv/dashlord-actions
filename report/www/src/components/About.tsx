import * as React from "react";

import CallOut from "@codegouvfr/react-dsfr/CallOut";

export const About: React.FC = () => (
  <CallOut className="fr-mb-3w" title="À propos">
    DashLord est <strong>100% open source</strong>, vous pouvez y contribuer en
    créant de la documentation, en proposant du code, ou encore en partageant
    vos questions{" "}
    <a
      href="https://github.com/socialgouv/dashlord"
      target="_blank"
      rel="noopener noreferrer"
    >
      sur le repository github
    </a>
    .
    <br /> <br />
    L&apos;objectif de DashLord est d&apos;identifier des points
    d&apos;attention mais aussi de proposer des solutions actionnables.
    <br /> <br />
    DashLord est né à la{" "}
    <a
      href="https://fabrique.social.gouv.fr"
      target="_blank"
      rel="noopener noreferrer"
    >
      {" "}
      Fabrique des ministères sociaux
    </a>{" "}
    pour répondre aux besoins d&apos;évaluation et de mise en oeuvre des bonnes
    pratiques de développement web.
    <br /> <br />
    Vous pouvez facilement créer votre propre DashLord pour vos URLs en suivant{" "}
    <a
      href="https://github.com/SocialGouv/dashlord/blob/main/README.md"
      target="_blank"
      rel="noopener noreferrer"
    >
      les instructions sur le GitHub
    </a>
    .
  </CallOut>
);
