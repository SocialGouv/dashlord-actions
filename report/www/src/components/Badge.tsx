import classNames from "classnames";
import React, { CSSProperties } from "react";
import { useRouter } from "next/router";

import styles from "./badge.module.scss";

type BadgeProps = {
  variant: string;
  className?: string;
  to?: string;
  style?: CSSProperties;
  external?: boolean;
  children: any;
};

const Badge: React.FC<BadgeProps> = ({
  children,
  external = false,
  variant,
  className,
  to,
  style,
}) => {
  const router = useRouter();
  return (
    <button
      type="button"
      className={classNames(className, styles[variant])}
      onClick={() => {
        if (to) {
          if (external) {
            window.open(to);
            return;
          }
          router.push(to);
        }
      }}
      style={style}
    >
      {children}
    </button>
  );
};

export default Badge;
