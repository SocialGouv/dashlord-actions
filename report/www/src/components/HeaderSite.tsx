import React from "react";
import { useRouter } from "next/router";
import uniq from "lodash.uniq";
import { Header } from "@codegouvfr/react-dsfr/Header";
import { symToStr } from "tsafe/symToStr";

import { headerFooterDisplayItem } from "@codegouvfr/react-dsfr/Display";
import { createComponentI18nApi } from "@codegouvfr/react-dsfr/i18n";

import dashlordConfig from "@/config.json";
import { sortByKey, isToolEnabled } from "../utils";

type HeaderSiteProps = {
  report: DashLordReport;
};

export const HeaderSite: React.FC<HeaderSiteProps> = ({ report }) => {
  const sortedReport = (report && report.sort(sortByKey("url"))) || [];
  const categories = uniq(
    sortedReport.filter((u) => u.category).map((u) => u.category)
  ).sort() as string[];
  const tags = uniq(
    sortedReport.filter((u) => u.category).flatMap((u) => u.tags)
  ).sort() as string[];
  const router = useRouter();

  const { t } = useTranslation();

  const betaStartups =
    (isToolEnabled("betagouv") &&
      sortedReport
        .filter((url) => url.betaId)
        .filter(
          (url, i, all) =>
            !all
              .map((u) => u.betaId)
              .slice(i + 1)
              .includes(url.betaId)
        )
        .map((url) => url.betaId)
        .sort()) ||
    [];

  const views = [
    isToolEnabled("wappalyzer") && {
      linkProps: {
        href: "/wappalyzer",
      },
      text: t("wappalyzer"),
      isActive: router.asPath === "/wappalyzer/",
    },
    isToolEnabled("updownio") && {
      linkProps: {
        href: "/updownio",
      },
      text: t("updownio"),
      isActive: router.asPath === "/updownio/",
    },
  ].filter(Boolean);

  return (
    <>
      <Header
        classes={{
          logo: (!!dashlordConfig.marianne && "auto") || "hidden",
        }}
        brandTop={dashlordConfig.entity}
        operatorLogo={
          dashlordConfig.operator &&
          (typeof dashlordConfig.operator.logo === "string"
            ? {
                orientation: "horizontal",
                imgUrl: dashlordConfig.operator.logo,
                alt: `Logo ${dashlordConfig.operator.name}`,
              }
            : {
                orientation: dashlordConfig.operator.logo.direction,
                imgUrl: dashlordConfig.operator.logo.src,
                alt: `Logo ${dashlordConfig.operator.name}`,
              })
        }
        homeLinkProps={{
          href: "/",
          title: dashlordConfig.title,
        }}
        quickAccessItems={[
          dashlordConfig.loginUrl && {
            text: t("login"),
            iconId: "fr-icon-lock-fill",
            linkProps: {
              href: dashlordConfig.loginUrl,
            },
          },
          headerFooterDisplayItem,
        ]}
        navigation={[
          {
            text: t("introduction"),
            linkProps: {
              href: "/intro",
            },
            isActive: router.asPath === "/intro/",
          },
          {
            text: t("dashboard"),
            linkProps: {
              href: "/",
            },
            isActive: router.asPath === "/",
          },
          categories.length > 1 && {
            text: t("categories"),
            menuLinks: [
              ...categories.map((category) => ({
                linkProps: {
                  href: `/category/${category}`,
                },
                text: category,
                isActive: router.asPath === `/category/${category}`,
              })),
            ],
            isActive: router.asPath.startsWith("/category/"),
          },
          tags.length > 1 && {
            text: t("tags"),
            menuLinks: [
              ...tags.map((tag) => ({
                linkProps: {
                  href: `/tag/${tag}`,
                },
                text: tag,
                isActive: router.asPath === `/tag/${tag}`,
              })),
            ],
            isActive: router.asPath.startsWith("/tag/"),
          },
          betaStartups.length > 1 && {
            text: t("startups"),
            menuLinks: betaStartups.map((startup) => ({
              linkProps: {
                href: `/startup/${startup}`,
              },
              text: startup,
              isActive: router.asPath === `/startup/${startup}`,
            })),
            isActive: router.asPath.startsWith("/startup/"),
          },
          views.length && {
            text: "Vues",
            menuLinks: views,
          },
          {
            text: t("about"),
            linkProps: {
              href: "/about",
            },
            isActive: router.asPath === "/about/",
          },
        ].filter(Boolean)}
        serviceTagline={dashlordConfig.description}
        serviceTitle={dashlordConfig.title}
      />
    </>
  );
};

const { useTranslation, addHeaderSiteTranslations } = createComponentI18nApi({
  componentName: symToStr({ HeaderSite }),
  frMessages: {
    /* spell-checker: disable */
    login: "Se connecter",
    introduction: "Introduction",
    dashboard: "Tableau de bord",
    categories: isToolEnabled("betagouv") ? "Incubateurs" : "Catégories",
    startups: "Startups",
    tags: "Tags",
    wappalyzer: "Technologies (wappalyzer)",
    updownio: "Disponibilité (updown.io)",
    about: "A propos",
  },
});

export { addHeaderSiteTranslations };

HeaderSite.displayName = symToStr({ HeaderSite });
