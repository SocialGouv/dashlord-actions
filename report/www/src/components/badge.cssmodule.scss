.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}

.success {
    background-color: var(--success);
}

.info {
    background-color: var(--info);
}

.warning {
    background-color: #FF6F4C;
}

.danger {
    background-color: var(--error);
}