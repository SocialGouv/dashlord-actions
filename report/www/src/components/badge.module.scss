.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}



.success, .success:hover {
    background-color: var(--success) !important;
}

.info, .info:hover {
    background-color: var(--info) !important;
}

.warning, .warning:hover {
    background-color: var(--warning) !important;
}

.danger, .danger:hover {
    background-color: var(--error) !important;
}