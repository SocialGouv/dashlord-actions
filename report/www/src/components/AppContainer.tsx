import * as React from "react";

import { Container, Alert } from "@dataesr/react-dsfr";

import { Switch, Route, useLocation, useParams } from "react-router-dom";

import { Dashboard } from "./Dashboard";
import { Trends } from "./Trends";
import { Intro } from "./Intro";
import { About } from "./About";
import { WappalyzerDashboard } from "./WappalyzerDashboard";
import Url from "./Url";

const report: DashLordReport = require("./report.json");
const trends: Trends = require("./trends.json");

type CategoryRouteProps = { report: DashLordReport };

// for some reason react-router `:url*` didnt work, use `*` only
interface CategoryParamTypes {
  category: string;
}

const CategoryRoute: React.FC<CategoryRouteProps> = (props) => {
  const params = useParams<CategoryParamTypes>();
  const category = window.decodeURIComponent(params.category);
  const urls = props.report.filter((u) => u.category === category);
  return (
    <>
      <br />
      {urls.length ? (
        <>
          <h3>
            {category} :{urls.length} urls
          </h3>
          <Dashboard report={urls} />
        </>
      ) : (
        <h3>Aucun URL associ√©e</h3>
      )}
    </>
  );
};

interface TagParamTypes {
  tag: string;
}

type TagRouteProps = { report: DashLordReport };

const TagRoute: React.FC<TagRouteProps> = (props) => {
  const params = useParams<TagParamTypes>();
  const tag = window.decodeURIComponent(params.tag);
  const urls = props.report.filter((u) => u.tags && u.tags.includes(tag));
  return (
    <>
      <br />
      <h3>
        {tag} :{urls.length} urls
      </h3>
      <Dashboard report={urls} />
    </>
  );
};

interface UrlParamTypes {
  "0": string;
}

type UrlRouteProps = { report: DashLordReport };

const UrlRoute: React.FC<UrlRouteProps> = (props) => {
  const params = useParams<UrlParamTypes>();
  const url = window.decodeURIComponent(params["0"]);
  const urlData = props.report.find((u) => u.url === url);
  if (!urlData) {
    return (
      <Alert
        type="error"
        title={`Impossible de trouver le rapport pour ${url}`}
      />
    );
  }
  return <Url url={url} report={urlData} />;
};

export const AppContainer = () => {
  const location = useLocation();
  return (
    <Container fluid={location.pathname === "/"}>
      <div role="main" className="fr-my-4w">
        <Switch>
          <Route path="/url/*">
            <UrlRoute report={report} />
          </Route>
          <Route path="/dashboard">
            <Dashboard report={report} />
          </Route>
          <Route path="/trends">
            <Trends trends={trends} />
          </Route>
          <Route path="/category/:category">
            <CategoryRoute report={report} />
          </Route>
          <Route path="/tag/:tag">
            <TagRoute report={report} />
          </Route>
          <Route path="/wappalyzer">
            <WappalyzerDashboard report={report} />
          </Route>
          <Route path="/intro">
            <Intro />
          </Route>
          <Route path="/about">
            <About />
          </Route>
          <Route path="/">
            <Dashboard report={report} />
          </Route>
        </Switch>
      </div>
    </Container>
  );
};
