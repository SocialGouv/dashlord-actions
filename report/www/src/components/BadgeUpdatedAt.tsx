import { formatDistanceToNow, differenceInDays, format } from "date-fns";
import { fr } from "@codegouvfr/react-dsfr";
import frLocale from "date-fns/locale/fr";

import { AlertProps } from "@codegouvfr/react-dsfr/Alert";
import Badge from "@codegouvfr/react-dsfr/Badge";
import { CSSProperties } from "react";

// badge that displya "since xxx" and appropriate colors
export const BadgeUpdatedAt = ({
  date,
  style,
}: {
  date: string;
  style: CSSProperties;
}) => {
  if (!date) return;
  const startDate = new Date(date);
  const lagInDays = differenceInDays(new Date(), startDate);
  let severity: AlertProps.Severity = "info";
  if (lagInDays > 60) {
    severity = "error";
  } else if (lagInDays > 30) {
    severity = "warning";
  }
  return (
    <Badge
      severity={severity}
      noIcon
      as="span"
      style={{ verticalAlign: "top", ...style }}
    >
      <i
        className={fr.cx("fr-icon-time-fill", "fr-icon--sm", "fr-mr-1v")}
        title={`Mis Ã  jour le ${format(new Date(startDate), "dd/M/yyyy")}`}
      />
      {formatDistanceToNow(new Date(startDate), {
        locale: frLocale,
      })}
    </Badge>
  );
};
