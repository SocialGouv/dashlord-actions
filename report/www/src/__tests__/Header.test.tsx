import { render } from "@testing-library/react";
import { HeaderSite } from "@/components/HeaderSite";

it("Should render HeaderSite", () => {
  const { container } = render(<HeaderSite />);
  expect(container).toMatchSnapshot();
});
