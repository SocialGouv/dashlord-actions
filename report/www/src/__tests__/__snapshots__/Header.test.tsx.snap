// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render HeaderSite 1`] = `
<div>
  <button
    aria-controls="fr-theme-modal"
    aria-hidden="true"
    class="fr-btn fr-hidden"
    data-fr-opened="false"
    id="fr-theme-modal-hidden-control-button"
    tabindex="-1"
    type="button"
  >
     
  </button>
  <dialog
    aria-labelledby="fr-modal-title-fr-theme-modal"
    class="fr-modal"
    data-fr-concealing-backdrop="true"
    id="fr-theme-modal"
    role="dialog"
  >
    <div
      class="fr-container fr-container--fluid fr-container-md"
    >
      <div
        class="fr-grid-row fr-grid-row--center"
      >
        <div
          class="fr-col-12 fr-col-md-6 fr-col-lg-4"
        >
          <div
            class="fr-modal__body"
          >
            <div
              class="fr-modal__header"
            >
              <button
                aria-controls="fr-theme-modal"
                class="fr-btn--close fr-btn"
                title="Fermer"
                type="button"
              >
                Fermer
              </button>
            </div>
            <div
              class="fr-modal__content"
            >
              <h1
                class="fr-modal__title"
                id="fr-modal-title-fr-theme-modal"
              >
                Paramètres d'affichage
              </h1>
              <div
                class="fr-display"
              >
                <div
                  class="fr-form-group"
                >
                  <fieldset
                    class="fr-fieldset"
                  >
                    <legend
                      class="fr-fieldset__legend fr-text--regular"
                    >
                      Choisissez un thème pour personnaliser l'apparence du site.
                    </legend>
                    <div
                      class="fr-fieldset__content"
                    >
                      <div
                        class="fr-radio-group fr-radio-rich"
                      >
                        <input
                          id=":r0:-light"
                          name="fr-radios-theme"
                          type="radio"
                          value="light"
                        />
                        <label
                          class="fr-label"
                          for=":r0:-light"
                        >
                          Thème clair
                        </label>
                        <div
                          class="fr-radio-rich__img"
                        >
                          <svg
                            aria-hidden="true"
                            height="80px"
                            viewBox="0 0 80 80"
                            width="80px"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <use
                              class="fr-artwork-decorative"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/light.svg#artwork-decorative"
                            />
                            <use
                              class="fr-artwork-minor"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/light.svg#artwork-minor"
                            />
                            <use
                              class="fr-artwork-major"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/light.svg#artwork-major"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="fr-radio-group fr-radio-rich"
                      >
                        <input
                          id=":r0:-dark"
                          name="fr-radios-theme"
                          type="radio"
                          value="dark"
                        />
                        <label
                          class="fr-label"
                          for=":r0:-dark"
                        >
                          Thème sombre
                        </label>
                        <div
                          class="fr-radio-rich__img"
                        >
                          <svg
                            aria-hidden="true"
                            height="80px"
                            viewBox="0 0 80 80"
                            width="80px"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <use
                              class="fr-artwork-decorative"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/dark.svg#artwork-decorative"
                            />
                            <use
                              class="fr-artwork-minor"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/dark.svg#artwork-minor"
                            />
                            <use
                              class="fr-artwork-major"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/dark.svg#artwork-major"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="fr-radio-group fr-radio-rich"
                      >
                        <input
                          id=":r0:-system"
                          name="fr-radios-theme"
                          type="radio"
                          value="system"
                        />
                        <label
                          class="fr-label"
                          for=":r0:-system"
                        >
                          Système.
                          <span
                            class="fr-hint-text"
                          >
                            Utilise les paramètres système.
                          </span>
                        </label>
                        <div
                          class="fr-radio-rich__img"
                        >
                          <svg
                            aria-hidden="true"
                            height="80px"
                            viewBox="0 0 80 80"
                            width="80px"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <use
                              class="fr-artwork-decorative"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/system.svg#artwork-decorative"
                            />
                            <use
                              class="fr-artwork-minor"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/system.svg#artwork-minor"
                            />
                            <use
                              class="fr-artwork-major"
                              xlink:href="/node_modules/@codegouvfr/react-dsfr/dsfr/artwork/system.svg#artwork-major"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </dialog>
  <header
    class="fr-header"
    id="fr-header"
    role="banner"
  >
    <div
      class="fr-header__body"
    >
      <div
        class="fr-container"
      >
        <div
          class="fr-header__body-row"
        >
          <div
            class="fr-header__brand fr-enlarge-link"
          >
            <div
              class="fr-header__brand-top"
            >
              <div
                class="fr-header__logo auto"
              >
                <p
                  class="fr-logo"
                >
                  Ministères sociaux
                </p>
              </div>
              <div
                class="fr-header__operator"
              >
                <a
                  href="/"
                  title="DashLord Fabrique des ministères sociaux"
                >
                  <img
                    alt="Logo Fabrique des ministères sociaux"
                    class="fr-responsive-img"
                    src="https://avatars.githubusercontent.com/u/35456799?s=400"
                    style="max-width: 9.0625rem;"
                  />
                </a>
              </div>
              <div
                class="fr-header__navbar"
              >
                <button
                  aria-controls="header-menu-modal-fr-header"
                  aria-haspopup="menu"
                  class="fr-btn--menu fr-btn"
                  data-fr-opened="false"
                  id="fr-header-menu-button"
                  title="Menu"
                >
                  Menu
                </button>
              </div>
            </div>
            <div
              class="fr-header__service"
            >
              <a
                href="/"
                title="DashLord Fabrique des ministères sociaux"
              >
                <p
                  class="fr-header__service-title"
                >
                  DashLord Fabrique des ministères sociaux
                </p>
              </a>
              <p
                class="fr-header__service-tagline"
              >
                Bonnes pratiques techniques
              </p>
            </div>
          </div>
          <div
            class="fr-header__tools"
          >
            <div
              class="fr-header__tools-links"
            >
              <ul
                class="fr-btns-group"
              >
                <li>
                  <a
                    class="fr-btn fr-icon-lock-fill"
                    href="https://dashlord-full.fabrique.social.gouv.fr"
                    id="fr-header-quick-access-item--Seconnecter-0"
                  >
                    Se connecter
                  </a>
                </li>
                <li>
                  <button
                    aria-controls="fr-theme-modal"
                    class="fr-btn fr-icon-theme-fill"
                    data-fr-opened="false"
                    id="fr-header-quick-access-item--_objectObject_-1"
                  >
                    Paramètres d'affichage
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="fr-header-menu-button"
      class="fr-header__menu fr-modal"
      id="header-menu-modal-fr-header"
    >
      <div
        class="fr-container"
      >
        <button
          aria-controls="header-menu-modal-fr-header"
          class="fr-btn--close fr-btn"
          id="fr-header-mobile-overlay-button-close"
          title="Fermer"
        >
          Fermer
        </button>
        <div
          class="fr-header__menu-links"
        >
          <ul
            class="fr-btns-group"
          >
            <li>
              <a
                class="fr-btn fr-icon-lock-fill"
                href="https://dashlord-full.fabrique.social.gouv.fr"
                id="fr-header-quick-access-item--Seconnecter_mobile-0"
              >
                Se connecter
              </a>
            </li>
            <li>
              <button
                aria-controls="fr-theme-modal"
                class="fr-btn fr-icon-theme-fill"
                data-fr-opened="false"
                id="fr-header-quick-access-item--_objectObject__mobile-1"
              >
                Paramètres d'affichage
              </button>
            </li>
          </ul>
        </div>
        <nav
          aria-label="Menu principal"
          class="fr-nav"
          id="fr-header-main-navigation"
          role="navigation"
        >
          <ul
            class="fr-nav__list"
          >
            <li
              class="fr-nav__item"
            >
              <a
                class="fr-nav__link"
                href="/intro"
                id="fr-header-main-navigation-link-0"
              >
                Introduction
              </a>
            </li>
            <li
              class="fr-nav__item"
            >
              <a
                aria-current="page"
                class="fr-nav__link"
                href="/"
                id="fr-header-main-navigation-link-1"
              >
                Tableau de bord
              </a>
            </li>
            <li
              class="fr-nav__item"
            >
              <button
                aria-controls="fr-header-main-navigation-menu-2"
                aria-expanded="false"
                class="fr-nav__btn"
                id="fr-header-main-navigation-button-2"
              >
                Vues
              </button>
              <div
                class="fr-menu fr-collapse"
                id="fr-header-main-navigation-menu-2"
              >
                <ul
                  class="fr-menu__list"
                >
                  <li>
                    <a
                      class="fr-nav__link"
                      href="/wappalyzer"
                      id="fr-header-main-navigation-menu-2-link-Technologies_wappalyzer_-0"
                    >
                      Technologies (wappalyzer)
                    </a>
                  </li>
                  <li>
                    <a
                      class="fr-nav__link"
                      href="/updownio"
                      id="fr-header-main-navigation-menu-2-link-Disponibilit__updown_io_-1"
                    >
                      Disponibilité (updown.io)
                    </a>
                  </li>
                </ul>
              </div>
            </li>
            <li
              class="fr-nav__item"
            >
              <a
                class="fr-nav__link"
                href="/about"
                id="fr-header-main-navigation-link-3"
              >
                A propos
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
